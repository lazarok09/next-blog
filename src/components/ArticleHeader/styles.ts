import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.medium} + 1rem);
      }
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
    @media ${theme.media.lteMedium} {
      font-size: calc(${theme.font.sizes.small} + 0.2rem);
    }
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
