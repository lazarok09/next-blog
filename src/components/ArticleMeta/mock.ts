import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: 'aowea3i902103102',

    slug: 'lazaro-souza',
    displayName: 'LÃ¡zaro Souza',
  },
  categories: [
    {
      id: '123125129120',
      displayName: 'JS',
      slug: 'javascript',
    },
    {
      id: '123125129120125512',
      displayName: 'Tech',
      slug: 'tech',
    },
  ],
} as ArticleMetaProps;
