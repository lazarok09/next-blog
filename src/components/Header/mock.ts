import { HeaderProps } from '.';

export default {
  blogName: 'LÃ¡zaro',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: 'https://github.com/lazarok09.png',
  showText: true,
} as HeaderProps;
