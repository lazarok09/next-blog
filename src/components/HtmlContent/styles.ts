import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    p {
      margin: ${theme.spacings.xlarge} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
      text-decoration: none;
    }
    a:hover {
      filter: brightness(80%);
      text-decoration: underline;
    }
    pre, code{
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.xlarge} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
  }
    img {
        width: 100%;

    }
    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
      max-width: 100%;

    }
    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }
    .image-style-side {
      float: right;
      max-width: 50%;
      margin: 0 ${theme.spacings.medium};
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }
    ul,
    ol {
      margin: ${theme.spacings.xlarge};
    }
    .table {
      max-width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
    }
    table th,
    table td{
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
    }
    iframe {
    max-width: 100%;
    height: 50vh;
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: 0;
      }
    }
  `}
`;
