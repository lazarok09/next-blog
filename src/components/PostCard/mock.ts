import { data } from "../../api/data.json";

export default data.posts[0];
