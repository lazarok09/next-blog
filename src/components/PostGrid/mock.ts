import { data } from '../../api/data.json';

export default { posts: data.posts };
