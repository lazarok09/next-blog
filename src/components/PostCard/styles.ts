import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";
export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }

    & a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }
    & a:hover {
      color: ${theme.colors.secondary};
    }
    // when we hover this element & i want to change the behavier of the a tag
    &:hover a {
      color: ${theme.colors.secondary};
    }
    &:hover img {
      opacity: 0.8;
    }
  `}
`;
export const Cover = styled.img`
  ${() => css`
    max-width: 100%;
    height: 100%;
    max-height: 280px;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css`
    word-break: break-word;
  `}
`;
