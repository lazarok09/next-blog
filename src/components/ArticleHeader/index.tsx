import { StrapiImage } from '../../shared-types/StrapiImage';
import { ArticleMeta, ArticleMetaProps } from '../ArticleMeta';
import { Heading } from '../Heading';
import * as Styled from './styles';
export type ArticleHeaderProps = {
  id: string;
  title: string;
  excerpt: string;
  cover: StrapiImage;
} & ArticleMetaProps;

export const ArticleHeader = ({
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
}: ArticleHeaderProps) => {
  return (
    <Styled.Wrapper>
      <Heading size="huge">{title}</Heading>
      <Styled.Excerpt>{excerpt}</Styled.Excerpt>
      <Styled.Cover alt={cover.alternativeText} src={cover.url} />
      <ArticleMeta
        author={author}
        categories={categories}
        createdAt={createdAt}
      />
    </Styled.Wrapper>
  );
};
export default ArticleHeader;
