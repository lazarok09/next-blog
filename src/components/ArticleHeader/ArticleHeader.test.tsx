import { screen } from '@testing-library/react';
import ArticleHeader, { ArticleHeaderProps } from '.';
import { renderTheme } from '../../styles/render-theme';
import { formatDate } from '../../utils/formatDate';
import mock from './mock';
const props: ArticleHeaderProps = mock;
describe('<ArticleHeader />', () => {
  it('should render heading, excerpt, cover img and meta', () => {
    const { container } = renderTheme(<ArticleHeader {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: props.title })).toBeInTheDocument();
    expect(screen.getByText(props.excerpt)).toBeInTheDocument();
    expect(screen.getByText(formatDate(props.createdAt))).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
