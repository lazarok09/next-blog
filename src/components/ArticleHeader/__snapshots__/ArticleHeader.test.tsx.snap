// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader /> should render heading, excerpt, cover img and meta 1`] = `
.c5 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c5 .categories span::after {
  content: ', ';
}

.c5 .categories span:last-child::after {
  content: '';
}

.c5 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(200%);
  filter: brightness(200%);
}

.c2 {
  color: #000000;
  line-height: 1.2;
  font-size: 6.4rem;
  text-transform: none;
}

.c0 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 .c1 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  margin: 2.4rem 0;
  font-size: 2.4rem;
}

.c4 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

@media (max-width:768px) {
  .c2 {
    font-size: 4.0rem;
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      Basics JavaScript stuff
    </h1>
    <p
      class="c3"
    >
      JavaScript is a cross-platform, object-oriented scripting language. The v8 by Chrome runs the Node and make JS be a performatic language.
    </p>
    <img
      alt="Basics JavaScript stuff"
      class="c4"
      src="https://res.cloudinary.com/lazarok09/image/upload/v1633798940/photo_1560415755_bd80d06eda60_ixid_Mnwx_Mj_A3f_DB_8_M_Hxwa_G90by1w_Y_Wdlf_Hx8f_G_Vuf_DB_8f_Hx8_and_ixlib_rb_1_2_f32360cc03.jpg"
    />
    <div
      class="c5"
    >
      <p>
        <span>
          Por 
        </span>
        <a
          href="/author/lazaro"
        >
          Lazaro
        </a>
        <span
          class="separetor"
        >
           | 
        </span>
        <time
          datetime="2020-04-20"
        >
          19 de abr. de 2020
        </time>
        <span
          class="separetor"
        >
           | 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/javascript"
            >
              JavaScript
            </a>
          </span>
        </span>
      </p>
    </div>
  </header>
</div>
`;
